tests/libostreetest.c

```diff
@@ -72,29 +72,88 @@ ot_test_setup_repo (GCancellable *cancellable,
   return g_steal_pointer (&ret_repo);
 }
 
+/* Determine whether we're able to relabel files. Needed for bare tests. */
+gboolean
+ot_check_relabeling (gboolean *can_relabel,
+                     GError  **error)
+{
+  g_auto(GLnxTmpfile) tmpf = { 0, };
+  if (!glnx_open_tmpfile_linkable_at (AT_FDCWD, ".", O_RDWR | O_CLOEXEC, &tmpf, error))
+    return FALSE;
+
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(GBytes) bytes = glnx_fgetxattr_bytes (tmpf.fd, "security.selinux", &local_error);
+  if (!bytes)
+    {
+      /* libglnx preserves errno */
```

Comment:
Hmm...not explicitly.  I think it likely mostly works today, but I'm certain that e.g. `glnx_null_throw_errno_prefix()` might invoke `malloc()` which can definitely make system calls... if we want to make that guarantee I think we'll need something more like systemd's `_PROTECT_ERRNO`.

Reply:
Nevermind, I forgot we do save it.